src/Nethermind/Nethermind.Network/P2P/NettyRlpStream.cs

```diff
@@ -97,7 +97,7 @@ public override int Position
 
         public override int Length => _buffer.ReadableBytes + (_buffer.ReaderIndex - _initialPosition);
 
-        public override bool HasBeenRead => _buffer.ReadableBytes > 0;
+        public override bool HasBeenRead => _buffer.ReadableBytes <= 0;
```

Comment:
Was it a bug before?
Do we have an unit test for that?

Reply:
Yes it was a bug. It was not used anywhere before in the code.
No there is no unittest for this, I guess.